### ISSUE TITLE
[LibOS] Targets of links are always assumed to be under chroot

### ISSUE THREAD

**Issue by dimakuv** [oscarlab/graphene#1387]
## Description of the problem:

Current logic in Graphene assumes that target file is under `chroot` (actual file system). This misses cases like `/dev/stdout` which is a link to `/proc/self/fd/1` (i.e., Graphene currently fails if target is under pseudo-FS). This is the offending line:

https://github.com/oscarlab/graphene/blob/80030b1157da371251f632e325df71e9454b4e55/LibOS/shim/src/fs/shim_namei.c#L394

## Expected results:

```
[     1] ---- shim_openat(AT_FDCWD,"/dev/stdout",O_WRONLY|O_CREAT|O_TRUNC,0666) = 0
```

## Actual results:

```
[     1] ---- shim_openat(AT_FDCWD,"/dev/stdout",O_WRONLY|O_CREAT|O_TRUNC,0666) = -21
```

## Additional information: 

Generally, the whole FS subsystem must be rewritten in Graphene.

**Comment by dimakuv** [comment]
UPDATE: It's Ok to not implement this (we don't really support symlinks in Graphene), so cleanly erroring out is fine.

**Comment by pwmarcz** [comment]
This appears to be fixed after #2453:

```
$ echo hello | graphene-direct busybox cat /dev/stdin
hello

$ graphene-direct busybox readlink /dev/stdin
/proc/self/fd/0

$ graphene-direct busybox readlink /proc/self/fd/0
/dev/tty
```
